src/main/java/io/anserini/docgen/Model.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * Anserini: An information retrieval toolkit built on Lucene
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.anserini.docgen;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.text.WordUtils;
+
+public class Model {
+  private Map<String, Object> defaults;
+  private Map<String, Map<String, Object>> collections;
+
+  public Map<String, Object> getDefaults() {
+    return defaults;
+  }
+
+  public void setDefaults(Map<String, Object> defaults) {
+    this.defaults = defaults;
+  }
+
+  public Map<String, Map<String, Object>> getCollections() {
+    return collections;
+  }
+
+  public void setCollections(Map<String, Map<String, Object>> collections) {
+    this.collections = collections;
+  }
+
+  private Object safeGet(Map<String, Object> obj, String key) {
+    return obj.getOrDefault(key, defaults.getOrDefault(key, null));
+  }
+
+  public String genIndexingCmd(String collection) {
```

Comment:
can we rename to more standard Java method names?
e.g., `generateIndexingCommand`
Yes more verbose, but also more clear...
Same for methods below...